netstate/auth/process_api_key.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package auth
+
+import  (
+	"os"
+	"net/http"
+	"crypto/subtle"
+	
+	"github.com/spf13/viper"
+)
+
+func InitializeHeaders() *http.Header {
+	// mock HTTP request headers
+
+	httpHeaders := http.Header {
+		"Accept-Encoding": {"gzip, deflate"},
+		"Accept-Language": {"en-US,en;q=0.9"},
+		"X-Api-Key":         {"12345"},
+		"Cache-Control":    {"max-age=0"},
+		"Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8"} ,
+		"Connection":      {"keep-alive"},
+	}
+	return &httpHeaders
+}
+
+func setEnv() {
+	// if flag is not set, we'll set the env
+	// desgined to match xApiKey
+
+	viper.SetEnvPrefix("API") 
+	os.Setenv("API_KEY", "12345")
+	viper.AutomaticEnv()
+}
+
+func ValidateApiKey(header string)(bool) {
+	// validates apikey with xApiKey header
+
+	apiKey := viper.GetString("key")
+
+	if len(apiKey) == 0 {
+		setEnv()
```

Comment:
i think it's fine to just return false here, and that way we don't need setEnv and that's up to viper to do.

Reply:
whoops, nevermind, you cover this case in the switch statement. yeah i think you can get rid of lines 40-44

Reply:
great! got it!